### DISCUSSION TITLE
How do you look at git history of a buffer?

### DISCUSSION THREAD

**Discussion by nipuna-perera** [LazyVim/LazyVim#1373] [Q&A]
Coming from Emacs, one of the features I miss the most here is the "Git Time-machine" ability where you can quickly view the contents of the current buffer according to git history. Sort of like going back in history according to git. I'd also like to interact with the buffer at a certain point in history and copy code if needed. 

How do I achieve that in lazyvim? I am exclusively using lazygit (floating) as set up OOTB with lazyVim and I am happy with pretty much everything else. Is there some sort of a plugin that enables this functionality? How are others doing this? Googling for this tell me it's sort of possible with vim.fugitive. But I want to avoid installing another git plugin just to get this single feature.

This is what i'm referring to - https://github.com/emacsmirror/git-timemachine

**Comment by dpetka2001** [comment]
Telescope has a `git_bcommits` feature and also if you use `diffview.nvim` it also has a `DiffviewFileHistory` feature. Try them out to see what fits your needs best.
